projects/plugins/crm/includes/ZeroBSCRM.Database.php

```diff
@@ -951,10 +951,23 @@ function zeroBSCRM_DB_canInnoDB(){
  */
 function jpcrm_database_engine( $pretty = false ) {
 	global $zbs;
-	if ( $pretty ) {
-		return $zbs->database_server_info['db_engine_label'];
+
+	// If $zbs is an object but database_server_info is not set, try to populate it.
+	if ( is_object( $zbs ) && ! isset( $zbs->database_server_info ) ) {
```

Comment:
Did you run into an issue with this? If `$zbs` is not an object, we should probably return early.

Related, `$zbs->database_server_info` will always be set if `$zbs` is an object, given that it's initialized as an empty object. It's also populated as early as possible (within `$zbs->verify_minimum_requirements()`).

In other words, if `$zbs` is an object, the rest of the checks are unnecessary and the fallback scenario isn't possible.